File: android/data/src/main/kotlin/com/bolsillo/data/seed/CategorySeed.kt
package com.bolsillo.data.seed

import com.bolsillo.data.db.entity.CategoryEntity

/**
 * Seed mirrors shared-assets/taxonomy/category-taxonomy.json. `colorToken` maps
 * into shared-assets/design/tokens.json → category (Article VI shared taxonomy).
 * New transactions default to `other`.
 */
object CategorySeed {
    const val DEFAULT_CATEGORY_ID = "other"

    val ALL: List<CategoryEntity> =
        listOf(
            cat("food", "category.food", "fork.knife", "restaurantes"),
            cat("food.groceries", "category.food.groceries", "cart", "mercado"),
            cat("food.restaurants", "category.food.restaurants", "takeoutbag.and.cup.and.straw", "restaurantes"),
            cat("food.coffee", "category.food.coffee", "cup.and.saucer", "cafe"),
            cat("transport", "category.transport", "car", "transporte"),
            cat("transport.fuel", "category.transport.fuel", "fuelpump", "transporte"),
            cat("transport.taxi", "category.transport.taxi", "car.side", "transporte"),
            cat("transport.transit", "category.transport.transit", "bus", "transporte"),
            cat("housing", "category.housing", "house", "vivienda"),
            cat("housing.rent", "category.housing.rent", "key", "vivienda"),
            cat("utilities", "category.utilities", "bolt", "servicios"),
            cat("utilities.electricity", "category.utilities.electricity", "bolt.fill", "servicios"),
            cat("utilities.internet", "category.utilities.internet", "wifi", "servicios"),
            cat("health", "category.health", "cross.case", "salud"),
            cat("leisure", "category.leisure", "gamecontroller", "ocio"),
            cat("education", "category.education", "book", "educacion"),
            cat("shopping", "category.shopping", "bag", "ropa"),
            cat("income", "category.income", "arrow.down.circle", "ingreso"),
            cat("transfer", "category.transfer", "arrow.left.arrow.right", "otros"),
            cat("other", "category.other", "ellipsis.circle", "otros"),
        )

    private fun cat(
        id: String,
        nameKey: String,
        icon: String,
        token: String,
    ) = CategoryEntity(id = id, nameKey = nameKey, icon = icon, colorToken = token)
}

File: ios/Packages/Bolsillo/Sources/BolsilloData/Seeds/CategorySeed.swift
import Foundation
import GRDB
import BolsilloDomain

public enum CategorySeed {
    // Derived from shared-assets/taxonomy/category-taxonomy.json (Article VI)
    public static let all: [BolsilloDomain.Category] = [
        BolsilloDomain.Category(id: "food",                  nameKey: "category.food",                   icon: "fork.knife",                    colorToken: "restaurantes"),
        BolsilloDomain.Category(id: "food.groceries",        nameKey: "category.food.groceries",         icon: "cart",                          colorToken: "mercado"),
        BolsilloDomain.Category(id: "food.restaurants",      nameKey: "category.food.restaurants",       icon: "takeoutbag.and.cup.and.straw",  colorToken: "restaurantes"),
        BolsilloDomain.Category(id: "food.coffee",           nameKey: "category.food.coffee",            icon: "cup.and.saucer",                colorToken: "cafe"),
        BolsilloDomain.Category(id: "transport",             nameKey: "category.transport",              icon: "car",                           colorToken: "transporte"),
        BolsilloDomain.Category(id: "transport.fuel",        nameKey: "category.transport.fuel",         icon: "fuelpump",                      colorToken: "transporte"),
        BolsilloDomain.Category(id: "transport.taxi",        nameKey: "category.transport.taxi",         icon: "car.side",                      colorToken: "transporte"),
        BolsilloDomain.Category(id: "transport.transit",     nameKey: "category.transport.transit",      icon: "bus",                           colorToken: "transporte"),
        BolsilloDomain.Category(id: "housing",               nameKey: "category.housing",                icon: "house",                         colorToken: "vivienda"),
        BolsilloDomain.Category(id: "housing.rent",          nameKey: "category.housing.rent",           icon: "key",                           colorToken: "vivienda"),
        BolsilloDomain.Category(id: "utilities",             nameKey: "category.utilities",              icon: "bolt",                          colorToken: "servicios"),
        BolsilloDomain.Category(id: "utilities.electricity", nameKey: "category.utilities.electricity",  icon: "bolt.fill",                     colorToken: "servicios"),
        BolsilloDomain.Category(id: "utilities.internet",    nameKey: "category.utilities.internet",     icon: "wifi",                          colorToken: "servicios"),
        BolsilloDomain.Category(id: "health",                nameKey: "category.health",                 icon: "cross.case",                    colorToken: "salud"),
        BolsilloDomain.Category(id: "leisure",               nameKey: "category.leisure",                icon: "gamecontroller",                colorToken: "ocio"),
        BolsilloDomain.Category(id: "education",             nameKey: "category.education",              icon: "book",                          colorToken: "educacion"),
        BolsilloDomain.Category(id: "shopping",              nameKey: "category.shopping",               icon: "bag",                           colorToken: "ropa"),
        BolsilloDomain.Category(id: "income",                nameKey: "category.income",                 icon: "arrow.down.circle",             colorToken: "ingreso"),
        BolsilloDomain.Category(id: "transfer",              nameKey: "category.transfer",               icon: "arrow.left.arrow.right",        colorToken: "otros"),
        BolsilloDomain.Category(id: "other",                 nameKey: "category.other",                  icon: "ellipsis.circle",               colorToken: "otros"),
    ]

    public static func seed(_ db: BolsilloDatabase) throws {
        try db.queue.write { dbq in
            guard try CategoryRecord.fetchCount(dbq) == 0 else { return }
            for cat in all {
                try CategoryRecord.from(cat).insert(dbq)
            }
        }
    }
}
